packages/gateway/src/gateway.js

```diff
@@ -65,11 +69,16 @@ export async function gatewayGet(request, env, ctx) {
 
     ctx.waitUntil(
       (async () => {
+        const contentLengthMb = Number(
+          winnerGwResponse.response.headers.get('content-length')
+        )
+
         await Promise.all([
           storeWinnerGwResponse(request, env, winnerGwResponse),
           settleGatewayRequests(),
-          // Cache request URL in Cloudflare CDN
-          cache.put(request.url, winnerGwResponse.response.clone()),
+          // Cache request URL in Cloudflare CDN if smaller than CLOUDFLARE_MAX_OBJECT_SIZE
+          contentLengthMb < CLOUDFLARE_MAX_OBJECT_SIZE &&
```

Comment:
I assume the max is inclusive...

```suggestion
          contentLengthMb <= CLOUDFLARE_MAX_OBJECT_SIZE &&
```